workflow/artifacts/oss/oss.go

```diff
@@ -39,11 +51,84 @@ var (
 	bucketLogFilePrefix    = "bucket-log-"
 )
 
+type ossCredentials struct {
+	teaCred credentials.Credential
+}
+
+func (cred *ossCredentials) GetAccessKeyID() string {
+	value, err := cred.teaCred.GetAccessKeyId()
+	if err != nil {
+		log.Printf("get access key id failed: %+v", err)
+		return ""
+	}
+	return tea.StringValue(value)
+}
+
+func (cred *ossCredentials) GetAccessKeySecret() string {
+	value, err := cred.teaCred.GetAccessKeySecret()
+	if err != nil {
+		log.Printf("get access key secret failed: %+v", err)
+		return ""
+	}
+	return tea.StringValue(value)
+}
+
+func (cred *ossCredentials) GetSecurityToken() string {
+	value, err := cred.teaCred.GetSecurityToken()
+	if err != nil {
+		log.Printf("get access security token failed: %+v", err)
+		return ""
+	}
+	return tea.StringValue(value)
+}
+
+type ossCredentialsProvider struct {
+	cred credentials.Credential
+}
+
+func newOidcCredential(roleARN, oidcProviderARN, oidcTokenFilePath string) credentials.Credential {
+	config := new(credentials.Config).
+		SetType(oidcRoleArnMode).
+		SetRoleArn(roleARN).
+		SetOIDCProviderArn(oidcProviderARN).
+		SetOIDCTokenFilePath(oidcTokenFilePath).
+		SetRoleSessionName(roleSessionName)
+
+	oidcCredential, err := credentials.NewCredential(config)
+	if err != nil {
+		panic(err)
+	}
+	return oidcCredential
+}
+
+func (p *ossCredentialsProvider) GetCredentials() oss.Credentials {
+	return &ossCredentials{teaCred: p.cred}
+}
+
 func (ossDriver *ArtifactDriver) newOSSClient() (*oss.Client, error) {
 	var options []oss.ClientOption
 	if token := ossDriver.SecurityToken; token != "" {
 		options = append(options, oss.SecurityToken(token))
 	}
+	if ossDriver.UseSDKCreds {
+		// using default provider chain in sdk to get credential
+		log.Infof("Using default provider chain for OSS driver")
+		cred, err := credentials.NewCredential(nil)
+		if err != nil {
+			panic(err)
+		}
+		provider := &ossCredentialsProvider{cred: cred}
+		return oss.New(ossDriver.Endpoint, "", "",
+			oss.SetCredentialsProvider(provider))
+	} else if ossDriver.RoleARN != "" && ossDriver.OidcProviderARN != "" && ossDriver.OidcTokenFilePath != "" {
+		// using oidc provider for RRSA auth
+		log.Infof("Using oidc provider for OSS driver")
+		cred := newOidcCredential(ossDriver.RoleARN, ossDriver.OidcProviderARN, ossDriver.OidcTokenFilePath)
+		provider := &ossCredentialsProvider{cred: cred}
+		return oss.New(ossDriver.Endpoint, "", "",
+			oss.SetCredentialsProvider(provider))
+	}
+	//using ak sec
```

Comment:
Be more descriptive or remove the comment

Reply:
fixed